File: TP1/python/calculs.py
# Fichier: calculs.py
# Opérations mathématiques et bitwise avec structures de contrôle
# auteur: Pierre MOLY
# Exercice 1.8

def main():
    # lecture des valeurs saisies par l'utilisateur
    num1 = 14
    num2 = 6
    op = input(f"Entrez une opération entre {num1} et {num2}: ")

    result = None

    if op == '+':
        # addition des deux nombres
        result = num1 + num2
    elif op == '-':
        # soustraction des deux nombres
        result = num1 - num2
    elif op == '*':
        # multiplication des deux nombres
        result = num1 * num2
    elif op == '/':
        # vérification de la division par 0
        if num2 == 0:
            print("Division par 0 impossible")
            return
        # division des deux nombres
        result = num1 // num2
    elif op == '&':
        # opération bitwise AND
        result = num1 & num2
    elif op == '|':
        # opération bitwise OR
        result = num1 | num2
    elif op == '~':
        # opération bitwise NOT
        result = ~num1
    else:
        # opération non valide
        print("Opération non valide")
        return

    # affichage du résultat
    if op == '~':
        print(f"L'opération mathématique {op}{num1} = {result}")
    else:
        print(f"L'opération mathématique {num1} {op} {num2} = {result}")

if __name__ == "__main__":
    main()

File: TP1/c/calculs.c
/* Fichier: calculs.c
* Opérations mathématiques et bitwise avec structures de contrôle
* auteur: Pierre MOLY
* Exercice 1.8
*/

#include <stdio.h>

int main(){

    // lecture des valeurs saisies par l'utilisateur
    int num1 = 14;
    int num2 = 6;
    char op;
    printf("Entrez une opération entre %d et %d: ", num1, num2);
    scanf("%c", &op);
    int result;

    // switch pour déterminer l'opération à effectuer
    switch (op){
    case '+':
        result = num1 + num2;
        break;
    case '-':
        result = num1 - num2;
        break;
    case '*':
        result = num1 * num2;
        break;
    case '/':
        if (num2 == 0){
            printf("Division par 0 impossible\n");
            return 1;
        }
        result = num1 / num2;
        break;
    case '&':
        result = num1 & num2;
        break;
    case '|':
        result = num1 | num2;
        break;
    case '~':
        result = ~num1;
        break;
    default:
        printf("Opération non valide\n");
        return 1;
    }

    // affichage du résultat
    if (op ==  '~'){
        printf("L'operation mathématiqe %c%d = %d\n", op, num1, result);
    }
    else{
        printf("L'operation mathématiqe %d %c %d = %d\n", num1, op, num2, result);
    }
    return 0;
}